ToDoApp/Models/CoreDataManager.swift

```diff
@@ -0,0 +1,172 @@
+//
+//  CoreDataManager.swift
+//  ToDoApp
+//
+//  Created by 최용헌 on 2023/05/15.
+//
+
+import UIKit
+
+import CoreData
+
+//MARK: - To do 관리하는 매니저 (코어데이터 관리)
+final class CoreDataManager {
+  static let shared = CoreDataManager()
+  private init() {}
+  
+  // 앱 델리게이트, 앱이 실행될 때 하나의 앱을 관리해주는 객체, 앱의 라이프사이관리를 담당
+  let appDelegate = UIApplication.shared.delegate as? AppDelegate
+  
+  // CoreData에서 데이터 생성 병경 저장, 임시저장소
+  lazy var context = appDelegate?.persistentContainer.viewContext
+  
+  // 엔터티 이름 (코어데이터에 저장된 객체)
+  let modelName: String = "ToDoData"
+  
+  // MARK: - [Read] 코어데이터에 저장된 데이터 모두 읽어오기
+  func getToDoList() -> [ToDoData] {
+    var toDoList: [ToDoData] = []
+    // 임시저장소 있는지 확인
+    if let context = context {
+      let request = NSFetchRequest<NSManagedObject>(entityName: self.modelName)
+      // 정렬순서를 정해서 넘겨줌
+      let dateOrder = NSSortDescriptor(key: "date", ascending: true)
+      request.sortDescriptors = [dateOrder]
+      
+      do {
+        // 임시저장소에서 데이터 가져오기 (fetch메서드)
+        if let fetchedToDoList = try context.fetch(request) as? [ToDoData] {
+          toDoList = fetchedToDoList
+        }
+      } catch {
+        print("가져오는 것 실패")
+      }
+    }
+    
+    return toDoList
+  }
```

Comment:
이런 에러가 발생할 수 있는 작업을 다룰 때에는 UI 레이어에서 에러가 발생했는지 여부에 따라 이벤트를 처리할 수 있도록 throwing 함수로 만들어주면 좋을 것 같아

`catch`에서 `print`로 끝내는게 아니라 다시 한번 `throw`로 에러를 던져주는거지

```suggestion
  func getToDoList() throws -> [ToDoData] {
    var toDoList: [ToDoData] = []
    // 임시저장소 있는지 확인
    if let context = context {
      let request = NSFetchRequest<NSManagedObject>(entityName: self.modelName)
      // 정렬순서를 정해서 넘겨줌
      let dateOrder = NSSortDescriptor(key: "date", ascending: true)
      request.sortDescriptors = [dateOrder]
      
      do {
        // 임시저장소에서 데이터 가져오기 (fetch메서드)
        if let fetchedToDoList = try context.fetch(request) as? [ToDoData] {
          toDoList = fetchedToDoList
        }
      } catch {
        print("가져오는 것 실패")
        throw(error)
      }
    }
    
    return toDoList
  }
```

이런 느낌으로..!

그러면 UI 레이어에서

``` Swift
do {
  let todo = try todoManager.getToDoList()
} catch {
  // 에러 메시지 팝업
}
```

이렇게 에러가 발생했다는 걸 화면에 표현해줄 수 있으니까!